### PR TITLE
Module manager : Restore modules actions order

### LINKED ISSUE #28331: The module actions order have been modified
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

The modules' actions order have been modified in https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/src/Adapter/Module/AdminModuleDataProvider.php#L73

This update impacts the actions order in Module manager page
Capture d’écran de 2022-04-25 23-37-09

### Expected behavior

_No response_

### Steps to reproduce

Go to Modules > Modules manager

In 1.7.8, in the actions dropdown, for example uninstall came before disable

After the modification disable comes first

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by sowbiba**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Please see issue. Because the order matters, it have been restored
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28331

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by NeOMakinG — DISMISSED** [review]

File: src/Adapter/Module/AdminModuleDataProvider.php [hunk 1]
```diff
@@ -80,14 +80,14 @@ class AdminModuleDataProvider implements ModuleInterface
...
-        Module::ACTION_ENABLE,
         Module::ACTION_CONFIGURE,
-        Module::ACTION_DISABLE,
         Module::ACTION_UNINSTALL,
+        Module::ACTION_ENABLE,
+        Module::ACTION_DISABLE,
         Module::ACTION_ENABLE_MOBILE,
         Module::ACTION_DISABLE_MOBILE,
         Module::ACTION_RESET,
+        Module::ACTION_UPGRADE,
```
Comment (atomiix) [inline src/Adapter/Module/AdminModuleDataProvider.php]: I think `upgrade` should stay at the top. This action will be used as "primary" action when available

Comment (sowbiba) [reply src/Adapter/Module/AdminModuleDataProvider.php]: You're absolutely right. My bad

**Review by khouloudbelguith — COMMENTED** [review]
Hello @sowbiba,

I have a difference between This PR, develop without PR, and 178x.
**Module without configuration page**

**_With PR_**, we have this order
1- Uninstall
2- Disable
3- Disable Mobile
4- Reset

image
**_Without PR and 178x_**, we have this order

1. Disable
2. Uninstall
3. Disable Mobile
4. Reset 

image


**Module with Upgrade button**

**_With PR_**, we have this order

1. Configure
2. Uninstall
3. Disable
4. Disable Mobile
5. Reset
6. Upgrade
image

**_Without PR_**, we have this order

1. Upgrade
2. Configure
3. Disable
4. Uninstall
5. Disable Mobile
6. Reset 
image

**_178x_**, we have this order

1. Upgrade
2. Uninstall
3. Disable
4. Disable Mobile
5. Rest
6. Configure 
image

PS: we have an issue https://github.com/PrestaShop/PrestaShop/issues/17137

Enable **Multistore** & Create a new shop, when **all shops** selected

**_WithPRandwithoutPR_**, we have this order

1. Configure
2. Uninstall 
3. Reset 
image

**_178x_**, we have this order

Configure
Uninstall
Disable
Disable Mobile
Reset

image

Ping @PrestaShop/product-team, I need your confirmation

Thanks!

**Comment by sowbiba** [comment]
Thanks @khouloudbelguith 
There is a constant, Uninstall must come before Disable
But I need confirmation about the configure position. Without update available it is in first position and with update in last 🤔

**Comment by kpodemski** [comment]
@sowbiba 

> There is a constant, Uninstall must come before Disable

Why? Isn't it better to allow people to disable the module easier than uninstalling it?

**Comment by sowbiba** [comment]
Ow, I meant there is a constant on what @khouloudbelguith described. I will do whatever the product team decide 😀
My opinion? You're right, disable is more radical than uninstall

**Review by matthieu-rolland — DISMISSED** [review]
Seems like this PR is doing what is asked in the issue

any more questions for the PM team @sowbiba, concerning the overall order of all actions?
Maybe it should be asked in the issue then 👍 

LGTM

**Comment by eshraw** [comment]
Hello @sowbiba thank you for the PR and thank you @khouloudbelguith for checking every possible situation.

In regard to #[25314](https://github.com/PrestaShop/PrestaShop/pull/25314) here is the proposed order :
- Module::ACTION_INSTALL,
- Module::ACTION_CONFIGURE,
- Module::ACTION_UPGRADE,
- Module::ACTION_ENABLE,
- Module::ACTION_ENABLE_MOBILE,
- Module::ACTION_DISABLE,
- Module::ACTION_DISABLE_MOBILE,
- Module::ACTION_RESET,
- Module::ACTION_UNINSTALL,

Feel free to ping me if you have any more question regarding this subject.

**Comment by atomiix** [comment]
@eshraw I can be wrong but with this order, configuration will be the main action even when the module is disabled. Wouldn't it be easier to see that the module is disabled by setting the enable action before the configuration?

**Comment by eshraw** [comment]
Thanks for the good catch @atomiix. Indeed we should in fact have this order to avoid masking the fact that a module is disabled : 

- Module::ACTION_INSTALL,
- Module::ACTION_ENABLE,
- Module::ACTION_CONFIGURE,
- Module::ACTION_UPGRADE,
- Module::ACTION_ENABLE_MOBILE,
- Module::ACTION_DISABLE,
- Module::ACTION_DISABLE_MOBILE,
- Module::ACTION_RESET,
- Module::ACTION_UNINSTALL,

cc @sowbiba, sorry for yet another change in the list :/

**Comment by atomiix** [comment]
@eshraw I would suggest this order:

- Module::ACTION_INSTALL
- Module::ACTION_ENABLE
- Module::ACTION_CONFIGURE
- Module::ACTION_UPGRADE
- Module::ACTION_ENABLE_MOBILE
- Module::ACTION_DISABLE_MOBILE
- Module::ACTION_DISABLE
- Module::ACTION_RESET
- Module::ACTION_UNINSTALL

because otherwise, when the module is enabled, `enable_mobile` will be **before `disable`** when mobile is disabled but `disable_mobile` will be **after `disable`** when mobile is enabled 😄

**Comment by eshraw** [comment]
Thank you @atomiix. Indeed, let's go with your suggestion, we need the consistency.

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by khouloudbelguith — APPROVED** [review]
Hello @sowbiba,

It is ok :heavy_check_mark: 

**Module without configuration page**
**_Without upgrade_**
Disable mobile
Disable
Reset
Uninstall
**_Disabled_**
Enable
Reset
Uninstall
**_EnabledAndMobileDisabled_**
Enable mobile
Disable
Reset
Uninstall

------------------------------------------------

**_With upgrade_**
Upgrade
Disable mobile
Disable
Reset
Uninstall
**_Disabled_**
Enabled
Upgrade
Reset
Uninstall
**_EnabledAndMobileDisabled_**
Upgrade
Enable mobile
Disable
Reset
Uninstall

-----------------------------------------

**Module with configuration page**
**_Without upgrade_**
Configure
Disable Mobile
Disable
Reset
Uninstall
**_Disabled_**
Enable
Configure
Reset 
Uninstall
**_EnabledAndMobileDisabled_**
Configure
Enable Mobile
Disable
Reset
Uninstall

-------------------------------------

**_With upgrade_**
Configure
Upgrade
Disable Mobile
Disable
Reset
Uninstall
**_Disabled_**
Enable
Configure
Upgrade
Reset
Uninstall
**_EnabledAndMobileDisabled_**
Configure
Upgrade
Enable Mobile
Disable
Reset 
Uninstall

---------------------------

**When All shops selected**
**Module without configuration page**
**_Without upgrade_**
Reset
Uninstall
**_With upgrade_**
Upgrade
Reset
Uninstall

-----------------------------------------

**Module with configuration page**
**_Without upgrade_**
Configure
Reset
Uninstall
**_With upgrade_**
Configure
Upgrade
Reset
Uninstall

https://user-images.githubusercontent.com/16067358/166983871-b2d656ac-03df-48e2-8534-4d37abcd3b73.mp4

PS: I reproduced this issue: https://github.com/PrestaShop/PrestaShop/issues/28433 when testing this PR but not reproduced also on develop.

PS: in the BO > Design > Theme & LOgo page > Pages Configuration tab, this order is not applied 
image
But it is related to the **Theme Customization** module.

Thanks!

**Comment by Progi1984** [comment]
Thanks @sowbiba & @khouloudbelguith

**Pull request merged**